### PR TITLE
[DDCI-174] Vocabulary Services - Viewing term definitions (Backend)

### PR DESCRIPTION
**Pull request by MarkCalvert**
This PR is for Update controlled vocabulary service to handle definitions (Backend) (https://it-partners.atlassian.net/browse/DDCI-194)

### CODE REVIEW

File: ckanext/vocabulary_services/logic/action/create.py [hunk 1]
```diff
@@ -77,9 +79,11 @@ def vocabulary_service_term_upsert(context, data_dict):
                 if (existing_term.label == label) and (existing_term.uri != uri):
                     # If label is the same but uri is different, let's create them.
                     vocabulary_service_term_create(context, data_dict)
-                elif (existing_term.label != label) or (existing_term.uri == uri):
+                elif (existing_term.label != label or existing_term.definition != definition) and existing_term.uri == uri:
```
Comment (MarkCalvert) [inline ckanext/vocabulary_services/logic/action/create.py:82]: @salsa-nathan  I updated the logic here to match the comment 'Update the term label if the URI is the same and label different.
Not sure if this was a bug or intended but I updated the logic from `or (existing_term.uri == uri)` to `and (existing_term.uri == uri)`

File: ckanext/vocabulary_services/logic/action/get.py [hunk 2]
```diff
@@ -78,12 +78,14 @@ def csiro_vocabulary_terms(context, data_dict):
                         term = response['@graph'][i]
                         uri = term.get('@id', None)
                         label = term.get('rdfs:label', None)
+                        definition = term.get('dct:definition', None) or term.get('dct:description', None)
```
Comment (MarkCalvert) [inline ckanext/vocabulary_services/logic/action/get.py]: @salsa-nathan Is the logic correct? If there is a definition, use the definition value and fallback to using description value if the definition does not exist.

Comment (salsa-nathan) [reply ckanext/vocabulary_services/logic/action/get.py]: @MarkCalvert I think `term.get('dct:definition', None)` might need to be changed to `term.get('skos:definition', None)` based on what I see in the JSON coming from CSIRO, e.g.

Snippet from http://registry.it.csiro.au/def/isotc211/MD_ClassificationCode?_format=jsonld :
```
{
    "@id" : "http://registry.it.csiro.au/def/isotc211/MD_ClassificationCode/secret",
    "@type" : [ "skos:Concept", "http://def.isotc211.org/iso19115/-1/2014/ConstraintInformation#MD_ClassificationCode" ],
    "dct:description" : "kept or meant to be kept private, unknown, or hidden from all but a select group of people",
    "rdfs:isDefinedBy" : {
      "@id" : "http://standards.iso.org/iso/19115/-1/ed-1/en/"
    },
    "rdfs:label" : "secret",
    "skos:definition" : "kept or meant to be kept private, unknown, or hidden from all but a select group of people",
    "skos:inScheme" : {
      "@id" : "http://def.isotc211.org/iso19115/-1/2014/ConstraintInformation/code/MD_ClassificationCode"
    },
```

File: ckanext/vocabulary_services/logic/action/create.py (hunk 1 above)
Comment (salsa-nathan) [reply ckanext/vocabulary_services/logic/action/create.py:82]: @MarkCalvert - yes I think the change you made makes sense

File: ckanext/vocabulary_services/logic/action/get.py (hunk 2 above)
Comment (MarkCalvert) [reply ckanext/vocabulary_services/logic/action/get.py]: Good spotting

**Review by salsa-nathan — APPROVED** [review]

**Pull request merged**